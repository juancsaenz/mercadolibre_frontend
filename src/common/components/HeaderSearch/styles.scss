@import '../../themes/colors.scss';

.header-root {
  background: $ml-primary;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding-left: 1em;

  .logo {
    height: 35px;
    position: relative;
    width: auto;
    padding: 10px;
  }

  .searchBar {
    display: flex;
    z-index: 5;
    padding-left: 1em;

    .ant-select {
      width: 100%;
    }

    input {
      border-radius: 0.3em 0 0 0.3em!important;
      box-shadow: none !important;
      border: none;
      height: 38px;
      width: 600px;
      padding-left: 1em;
    }

    button {
      background-color: 'gray';
      border: none !important;
      color: 'black';
      height: 38px;
      width: 38px;
      cursor: pointer;
      border-radius: 0;
      font-size: 0.85em;

      .text {
        font-family: 'Roboto', sans-serif;
      }
    }
  } 
}
