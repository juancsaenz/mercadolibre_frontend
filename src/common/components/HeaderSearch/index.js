/**
 * @author Camilo SÃ¡enz
 * @file index.jsx
 * @description HeaderSearch Component
 */

// Dependencies
import React from 'react';
import { Layout, Input } from 'antd';
import { withRouter, Link } from "react-router-dom";

// Assets
import LogoIcon from '../../assets/Logo_ML.png';

// Styles
import './styles.scss';

const { Header } = Layout;
const { Search } = Input;

/**
 * AppHeaderSearch Function component.
 * @return { Node } React node containing app header with search bar.
 */
export class HeaderSearch extends React.Component {
  constructor(props) {
    super(props);

    this.searchItems = this.searchItems.bind(this);
  }

  searchItems(query) {
    this.props.history.push(`/items?q=${query}`);
  }

  render() {
    return (
      <div className="header-root">
        <div>
          <img
            src={LogoIcon}
            className='logo'
          />
        </div>
        <div className="searchBar">
          <Search
            placeholder="Nunca dejes de buscar"
            enterButton
            className="searchInput"
            onSearch={this.searchItems}
          />
        </div>
      </div>
    );
  }
};

export default withRouter(HeaderSearch);
