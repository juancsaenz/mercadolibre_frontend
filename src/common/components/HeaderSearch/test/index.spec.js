/**
 * @author Camilo SÃ¡enz
 * @file index.spec.js
 * @description Unit test to HeaderSearch file
 */

// Dependencies
import React from 'react';
import { shallow } from 'enzyme';

// Component
import HeaderSearch from '../index.js';

describe('<HeaderSearch />', () => {
  it('should render the component', () => {
    const renderedComponent = shallow(<HeaderSearch />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
