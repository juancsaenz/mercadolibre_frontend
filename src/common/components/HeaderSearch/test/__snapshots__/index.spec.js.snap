// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderSearch /> should render the component 1`] = `ShallowWrapper {}`;
